### DISCUSSION TITLE
Multiple Blocklists on Trasmission 4.0

### DISCUSSION THREAD

**Discussion by Javihache** [transmission/transmission#5093] [Support] [answered]
I have been using multiple blocklists as .bin files inside the Application Support/Transmission/Blocklists folder on MacOS since as far as I can remember (2010). Since the update to Transmission 4.0 I've noticed that the Blocklists do not work as .bin files inside the folder. Only URLs work and I can only use one single URL. 

Is there a possibility to use multiple blocklists with Transmission 4?

**Comment by Javihache** [comment]
bump

**Comment by dantearmok** [comment]
I just played around with various blocklist configurations....

Transmission used to ship with a URL in that field.  Now it ships with nothing.   I dunno why.

When launched, Transmission 4.0.2 processes the file(s) in the blocklists folder.  Specifically, it converts the text files to .bin then it loads them.  Not sure if there's a limit there - I've got 4 lists (.bin files) loaded currently.

Of course, that doesn't do anything wrt updating them - for that you need a url.  Currently, I'm using this:

http://list.iblocklist.com/?list=bt_level1&fileformat=p2p&archiveformat=gz

Try that URL.  If you can identify additional ip's that should be on it, let us know.

hth,
- Dan.

**Reply by nevack** [reply]
> Transmission used to ship with a URL in that field. Now it ships with nothing. I dunno why.

Do you mean there was some blacklist url by default? I doubt it.

**Reply by dantearmok** [reply]
According to my notes, the original default URL was http://www.bluetack.co.uk/config/level1.gz.  It died in early January 2012.  I think Transmission 2.61 dumped the url in July.

**Comment by antoflav** [comment]
Good morning everyone,
I install vers. 4.0.2 of Transmission-gtk, from snapcraft.io. My OS is Ubuntu-Gnome, Xenial 16.04.7. Everything works fine, except for blocklist. In fact even if I tried all urls finding on network, when I click "update" button, the blocklist is loaded, then message "Blocklist updated!" is show and immediately after the message "blocklist  has 0 entries" is showed. Permissions for that file are correctly: 644 with owner set to myself. However if I deleting the file, it is automatically recreated after "update" clicking. The urls that I tried:

_- _http://list.iblocklist.com/?list=bt_level1&fileformat=p2p&archiveformat=gz_
_- https://github.com/Naunter/BT_BlockLists/raw/master/bt_blocklists.gz_
_- https://github.com/sahsu/transmission-blocklist/releases/download/1.0.0/blocklist.gz__

None of the above works. I try to manually download and decompress blocklist file and the file thus obtained is NOT empty. But in path "~/snap/transmission-community/current/.config/transmission" there is the file "blocklist" (without any extension) that is EMPTY (0 bytes).

Any suggestion is appreciated.

Kind regards.

**Reply by Naunter** [reply]
My `Transmission 4.0.2 (2a57b17031)` is installed via Flatpak, the blocklist works fine on it.

Since yours is installed via snap, you may want to check the `snap` file access permission. It may also related to the `apparmor`.

One way you can try is to check the logs:
1. Open a terminal console
2. Type the following command: `sudo tail -f /var/log/syslog | grep -Ei 'apparmor|snap'`
3. Go to your Tranmission and then click `Update` blocklist
4. Check the terminal to see if any logs reported

I don't have snap installed, so can't test it for you. 

![image](https://user-images.githubusercontent.com/17890335/230338842-98e9dbc4-b750-4566-a7dc-1e6782c78d3f.png)
![image](https://user-images.githubusercontent.com/17890335/230338998-d8c9f97e-2f2c-4f98-bc1a-9eed1ff49659.png)

**Comment by Naunter** [comment]
@Javihache Not sure if tranmission can do multiple blocklists, but you can combine multiple rules in one file and then use it.

**Reply by dantearmok** [reply, marked as answer]
Yes, Transmission can do multiple blocklists. 

Drop the new lists into the blocklist folder and relaunch Transmission.  T4 will see the new lists, convert them into a binary format, and load them.  The update interface is to do one list only.

hth,
- Dan.
